File: src/maze/algos/random_java/Algo.java
public class Algo {
    public void generate(Interface interface_) {
        for (int i = 0; i < interface_.getWidth(); i += 1) {
            for (int j = 0; j < interface_.getHeight(); j += 1) {
                for (int k = 0; k < 4; k += 1) {
                    char direction = new String("nesw").charAt(k);
                    interface_.setWall(
                        i,
                        j,
                        direction,
                        interface_.getRandomFloat() < .4
                    );
                }
            }
        }
    }
}

File: src/maze/algos/random_c++/Algo.cpp
#include "Algo.h"

#include <string>

void Algo::generate(Interface* interface) {
    std::string directions = "nesw";
    for (int x = 0; x < interface->getWidth(); x += 1) {
        for (int y = 0; y < interface->getHeight(); y += 1) {
            for (int i = 0; i < directions.size(); i += 1) {
                bool wallExists = interface->getRandomFloat() <= 0.40;
                interface->setWall(x, y, directions.at(i), wallExists);
            }
        }
    }
}
